ERCS/erc-6538.md

```diff
@@ -27,65 +27,61 @@ This contract defines an `ERC5564Registry` that stores the stealth meta-address
   function registerKeys(uint256 scheme, bytes memory stealthMetaAddress) external {
-    stealthMetaAddressOf[abi.encode(msg.sender)][scheme] = stealthMetaAddress;
+    stealthMetaAddressOf[msg.sender][scheme] = stealthMetaAddress;
   }
 
   /// @notice Sets the `registrant`s stealth meta-address for the given scheme.
   /// @param registrant Recipient identifier, such as an ENS name.
-  /// @param scheme An integer identifier for the stealth address scheme.
+  /// @param schemeId An integer identifier for the stealth address scheme.
   /// @param signature A signature from the `registrant` authorizing the registration.
   /// @param stealthMetaAddress The stealth meta-address to register.
   /// @dev MUST support both EOA signatures and EIP-1271 signatures.
   /// @dev MUST revert if the signature is invalid.
   function registerKeysOnBehalf(
     address registrant,
-    uint256 scheme,
+    uint256 schemeId,
     bytes memory signature,
     bytes memory stealthMetaAddress
   ) external {
-    // TODO If registrant has no code, spit signature into r, s, and v and call `ecrecover`.
-    // TODO If registrant has code, call `isValidSignature` on the registrant.
-  }
-
-  /// @notice Sets the `registrant`s stealth meta-address for the given scheme.
-  /// @param registrant Recipient identifier, such as an ENS name.
-  /// @param scheme An integer identifier for the stealth address scheme.
-  /// @param signature A signature from the `registrant` authorizing the registration.
-  /// @param stealthMetaAddress The stealth meta-address to register.
-  /// @dev MUST support both EOA signatures and EIP-1271 signatures.
-  /// @dev MUST revert if the signature is invalid.
-  function registerKeysOnBehalf(
-    bytes memory registrant,
-    uint256 scheme,
-    bytes memory signature,
-    bytes memory stealthMetaAddress
-  ) external {
-    // TODO How to best generically support any registrant identifier / name
-    // system without e.g. hardcoding support just for ENS?
+    bytes32 digest = _hashTypedDataV4(
+      keccak256(
+        abi.encode(TYPE_HASH, registrant, schemeId, stealthMetaAddress, nonceOf[registrant]++)
+      )
+    );
+    require(isValidSignatureNow(registrant, digest, signature), "Invalid signature");
+    stealthMetaAddressOf[registrant][schemeId] = stealthMetaAddress;
+    emit StealthMetaAddressSet(registrant, schemeId, stealthMetaAddress);
   }
 }
 ```
 
-Deployment is done using the keyless deployment method commonly known as Nick’s method, TODO continue describing this and include transaction data, can base it off the format/description used in [ERC-1820](./eip-1820.md) and [ERC-2470](./eip-2470.md).
+### Deployment Method
+
+This contract is going to be deployed using the CREATE2 method. A predetermined address will use Nick's deterministic deployment proxy address of 0x4e59b44847b379578588920ca78fbf26c0b4956c to deploy the contract.
+An empty string, "", is going to be used as the salt.
```

Comment:
Salt must be 32 bytes so an empty string is not valid. You can use all zeros, but we may want to mine a salt for a fun vanity address
```suggestion
This contract is going to be deployed using CREATE2 via the deterministic deployer at `0x4e59b44847b379578588920ca78fbf26c0b4956c` with a salt of TODO.
```

Reply:
Updated in the same commit as above:  https://github.com/garyghayrat/ERCs/pull/2/commits/0163a5a14a3b9517802953c5b6c7a5b68903cd4a